staging/src/k8s.io/client-go/openapi/openapitest/fake.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapitest
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/client-go/openapi"
+)
+
+// NewFileOpenAPIClient returns a pointer to a testing
+// FileOpenAPIClient, which parses the OpenAPI GroupVersion
+// files in the passed directory path. Example GroupVersion
+// OpenAPI filename for apps/v1 would look like:
+//
+//	apis__apps__v1_openapi.json
+//
+// The current directories housing these hard-coded GroupVersion
+// OpenAPI V3 files is at:
+//
+//	<K8S_ROOT>/api/openapi-spec/v3
+//
+// An example to invoke this function for the test files:
+//
+//	NewFileOpenAPIClient("api/openapi-spec/v3")
+//
+// This function will search all subdirectories for .json files.
+func NewFileOpenAPIClient(dir string) (*FileOpenAPIClient, error) {
```

Comment:
While I was writing the code snippet below, it came to me that `openapi.NewFileOpenAPIClient` does repeat the `openapi` twice. What about renaming this to `openapi.NewFileClient`?